src/TableRow.js

```diff
@@ -73,7 +73,12 @@ class TableRow extends React.Component {
   }
 
   shouldComponentUpdate(nextProps) {
-    return !!(this.props.visible || nextProps.visible);
+    const shallowCompare = (obj1, obj2) =>
+      Object.keys(obj1).length === Object.keys(obj2).length &&
+      Object.keys(obj1).every(key => obj1[key] === obj2[key]);
+    const x = !shallowCompare(this.props.record, nextProps.record);
```

Comment:
add a new boolean prop called 'pureRow'. If that is true, only then do shallow compare, otherwise normal thing

Reply:
we can also add visibility check when pureRow is true

- if currently its not visible, then return false
- if earlier it was not visible, but now its visible, we should return true without checking shallow compare